### ISSUE TITLE
crash on launched

### ISSUE THREAD

**Issue by isAAAc** [savonet/liquidsoap#2338]
**Describe the bug**
liquidsoap crashed each time i launch it since yersterday evening,
it works on the morning,  

i can't remember any upgrade during the day.    

OS version && liquidsoap version (liquidsoap installed via apt):
```
isaaac@krasheur:~$ cat /etc/debian_version && dpkg -l |grep liquidsoap
bookworm/sid
ii  liquidsoap                           2.0.2-1                           amd64        audio streaming language
```

command launched:
```
isaaac@krasheur:~$ liquidsoap radio.liq -v
```
content of radio.liq:
```
isaaac@krasheur:~$ cat radio.liq

#/usr/bin/liquidsoap
output.icecast(%vorbis,
 host = "<<NDD>", port = 8000,
 password = "<PASWD>", mount = "<MOUTPOINT>",
 url="https://<FQDN>/station",
 name="KrashboyZ BordeL Klub",
 genre="krcksckrrchsshSrhSkkKrrRRRR",
 description="La Radio des Cons Rennais",
 public=true,
 fallible=true,
 mksafe(playlist(mode="randomize","/mnt/disks/Musique/Musique classee")))

```

output:
```
Called from Evaluation.eval.ans in file "lang/evaluation.ml", line 206, characters 30-75
Called from Evaluation.apply.f in file "lang/evaluation.ml", line 250, characters 8-12
Called from Tutils.finalize in file "tools/tutils.ml", line 100, characters 12-16
Re-raised at Tutils.finalize in file "tools/tutils.ml", line 106, characters 4-38
Called from Evaluation.apply.f in file "lang/evaluation.ml", line 250, characters 8-12
Called from Evaluation.apply.f in file "lang/evaluation.ml", line 250, characters 8-12
Called from Evaluation.eval.ans.(fun) in file "lang/evaluation.ml", line 206, characters 59-70
Called from Stdlib__list.map in file "list.ml", line 92, characters 20-23
Called from Stdlib__list.map in file "list.ml", line 92, characters 32-39
Called from Evaluation.eval.ans in file "lang/evaluation.ml", line 206, characters 30-75
Called from Evaluation.apply.f in file "lang/evaluation.ml", line 250, characters 8-12
Called from Tutils.finalize in file "tools/tutils.ml", line 100, characters 12-16
Re-raised at Tutils.finalize in file "tools/tutils.ml", line 106, characters 4-38
Called from Evaluation.apply.f in file "lang/evaluation.ml", line 250, characters 8-12
Called from Evaluation.apply.f in file "lang/evaluation.ml", line 250, characters 8-12
```

thx for your help :crossed_fingers: :sos:

**Comment by smimram** [comment]
Can you also provide the line just above the output you pasted (should begin with `Raised ...`)? We cannot see the actual exception otherwise.

**Comment by isAAAc** [comment]
yes sure @smimram , 
here is the full output:
[liquidsoap_20220415_fulloutput.txt](https://github.com/savonet/liquidsoap/files/8495261/liquidsoap_20220415_fulloutput.txt)

thx for your help ;)

**Comment by smimram** [comment]
Great, so this is a stack overflow, which basically means that a function, somewhere, is calling itself indefinitely. How big is your playlist? My guess is that it is really big and you have been hit by #2162. Am I right? :)

**Comment by isAAAc** [comment]
yes, you got it ^^, folder in random read:
```
  236.9 GiB [######### ] /Musique classee
```

**Comment by isAAAc** [comment]
@smimram the fix you done un january is not in the debian testing packet version  2.0.2-1 ?

**Comment by isAAAc** [comment]
i have my answer, looks fixed in 2.0.3
but 2.0.3 is not in debian bookworm
everything is OK with https://github.com/savonet/liquidsoap/releases/download/v2.0.3/liquidsoap_2.0.3-debian-testing-2_amd64.deb installed , 
thx for your help and your work on liquidsoap @smimram <3

i close the issue

**Comment by smimram** [comment]
Unfortunately, the fix is not in 2.0.2, no :(
